src/freebsd/topology.c

```diff
@@ -0,0 +1,33 @@
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+#include <cpuinfo/log.h>
+#include <freebsd/api.h>
+
+
+
+struct cpuinfo_freebsd_topology cpuinfo_freebsd_detect_topology(void) {
+	int threads = 1;
+	size_t sizeof_threads = sizeof(threads);
+	if (sysctlbyname("hw.ncpu", &threads, &sizeof_threads, NULL, 0) != 0) {
+		cpuinfo_log_error("sysctlbyname(\"hw.ncpu\") failed: %s", strerror(errno));
+	} else if (threads <= 0) {
+		cpuinfo_log_error("sysctlbyname(\"hw.ncpu\") returned invalid value %d", threads);
+	}
+
+    int cores = threads / 2;
```

Comment:
This is not acceptable. The code must detect topology, not fake it.

Reply:
> This is not acceptable. The code must detect topology, not fake it.

Sorry, I changed to use the proper system calls.